### ISSUE TITLE
Feat: Option to configure SSL cert for curl

### ISSUE THREAD

**Issue by ficcdaf** [oflisback/obsidian-bridge.nvim#27]
**TL;DR** connecting `obsidian-bridge` to the REST API via. HTTPS/SSL should be much easier for users to set up. I propose either a setup option or environment variable which tells `curl` what to pass with the `--cacert` flag. This would not be a breaking change. I will fork this repository, implement the solution, and submit a PR myself; I am creating this issue to also start a discussion and let folks weigh in on how they think this implementation should be done.

# Motivation

If you want to use the SSL encrypted port on the Local REST API plugin, you need to "trust" the certificate generated by the plugin. This is fairly easy to set up in a web browser (which is also the main use case of the REST API plugin). 

However, for use with obsidian-bridge, I really struggled. Since bridge uses `curl` under the hood, it's as trivial as passing a CLI flag pointing to the certificate file to all `curl` calls. `obsidian-bridge` does not expose this as a setup option, however. So, if a user wants to use SSL, they need to either modify this plugin's source code themselves, which is time consuming and not very noob friend, *or* figure out how to "trust" the certificate system-wide, which differs greatly among operating systems, and is also quite technically involved. 

Either option is a lot of work and research for the user to set up, and it's much easier to just fall back on the default option of an unencrypted connection, which is bad security practice, even despite the fact that it's only a localhost connection.

# Proposed Solution

A configuration option that can be passed to the `setup` function when the plugin is loaded. This option could contain a string which is the path to the certificate file that users can copy from the Local REST API interface. Alternatively, an environment variable could also contain this path. `obsidian-bridge` would pass this string as the value of the `--cacert` flag of `curl`. 

**Proposed default:** If the configuration option is `nil`, or the environment variable is not set (depending on which implementation is chosen), then the plugin should omit the `--cacert` flag in all `curl` calls, retaining its previous behavior. Therefore, this would not be a breaking change, and users would not need to update their configuration upon updating this plugin if they want to keep using unencrypted HTTP traffic. 

# Implementation

I will fork this repository, implement the change, and submit a PR myself. I'm creating this issue for reference, as well as to get feedback on this idea. So @oflisback don't worry too much about this, because this is less of a "feature request" and more of an "I'm going to implement this feature and submit a PR, is my proposed approach fine, should I change it somehow, or should I just not bother because you don't want this feature in the plugin?"

I look forward to hearing what you all think about this.

**Comment by oflisback** [comment]
Hey @ficcdaf that sounds great and I think a config option is better than an env variable. Happy to test and merge a PR for this if you're up for it!
